src/test/mir-opt/const_prop/discriminant.rs

```diff
@@ -31,7 +31,7 @@ fn main() {
 // START rustc.main.ConstProp.after.mir
 //  bb0: {
 //      ...
-//      _3 = const Scalar(0x01) : std::option::Option<bool>;
+//      _3 = const {0x1: std::option::Option::<bool>};
```

Comment:
I liked the padding-to-type-size that we did before here.

Reply:
No this was not resolved. I meant `0x01` -- padding the number to the size of the type.

Reply:
oh huh... I thought I was doing that, investigating

Reply:
-.- I was doing that, but I halved the amount of zeros that should be added, because they are twice as many as bytes

Reply:
`<Scalar as Debug>::fmt` also does this. Not sure if it's worth trying to deduplicate that.

Reply:
`Scalar` has no type information. Any deduplication we'd do will end up being more complex than the little bit of code we have there